components/remote-environment-broker/internal/mapping/controller.go

```diff
@@ -189,7 +196,7 @@ func (c *Controller) processItem(key string) error {
 			return err
 		}
 		if c.manageNsBrokers {
-			if err := c.ensureNsBrokerNotRegistered(namespace); err != nil {
+			if err := c.ensureEnvMappings(namespace); err != nil {
```

Comment:
I don't like this method name. On the basis of its name, what is your assumption about its responsibilities?
I would expect that "ensureEnvMapping" creates EnvironmentMapping.
My proposal: `ensureNsBrokerNotRegisteredIfNoMappings`, maybe you will find a better one